mmv1/products/integrationconnectors/Connection.yaml

```diff
@@ -0,0 +1,1120 @@
+                - !ruby/object:Api::Type::String
+                  name: "kmsKeyName"
+                  description: |
+                    The [KMS key name] with which the content of the Operation is encrypted. The expected
+                    format: projects/*/locations/*/keyRings/*/cryptoKeys/*.
+                    Will be empty string if google managed.
+      - !ruby/object:Api::Type::String
+        name: "authKey"
+        description: |
+          The type of authentication configured.
+      - !ruby/object:Api::Type::NestedObject
+        name: "userPassword"
+        properties:
+          - !ruby/object:Api::Type::String
+            name: "username"
+            description: |
+              Username for Authentication.
+          - !ruby/object:Api::Type::NestedObject
+            name: "password"
+            description: |
+              Password for Authentication.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "secretVersion"
+                required: true
+                description: |
+                  The resource name of the secret version in the format,
+                  format as: projects/*/secrets/*/versions/*.
+        exactly_one_of:
+          - auth_config.0.user_password
+          - auth_config.0.oauth2_jwt_bearer
+          - auth_config.0.oauth2_client_credentials
+          - auth_config.0.ssh_public_key
+          - auth_config.0.oauth2_auth_code_flow
+      - !ruby/object:Api::Type::NestedObject
+        name: "oauth2JwtBearer"
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: "clientKey"
+            description: |
+              Secret version reference containing a PKCS#8 PEM-encoded private key associated with the Client Certificate.
+              This private key will be used to sign JWTs used for the jwt-bearer authorization grant.
+              Specified in the form as: projects/*/secrets/*/versions/*.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "secretVersion"
+                required: true
+                description: |
+                  The resource name of the secret version in the format,
+                  format as: projects/*/secrets/*/versions/*.
+          - !ruby/object:Api::Type::NestedObject
+            name: "jwtClaims"
+            description: |
+              JwtClaims providers fields to generate the token.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "issuer"
+                description: |
+                  Value for the "iss" claim.
+              - !ruby/object:Api::Type::String
+                name: "subject"
+                description: |
+                  Value for the "sub" claim.
+              - !ruby/object:Api::Type::String
+                name: "audience"
+                description: |
+                  Value for the "aud" claim.
```

Comment:
Do we need `exactly_one_of` for these subfields? basically can we set more than one of these values? if so, we probably need `at_least_one_of`, otherwise, `exactly_one_of`

Reply:
All are required.  Modified.